src/irmin-pack/irmin_pack.ml

```diff
@@ -254,6 +268,15 @@ module Atomic_write (K : Irmin.Type.S) (V : Irmin.Hash.S) = struct
   let watch t = W.watch t.w
 
   let unwatch t = W.unwatch t.w
+
+  let close t =
+    t.counter <- t.counter - 1;
+    if t.counter = 0 && Lwt_mutex.is_empty t.lock then (
```

Comment:
you should probably try to take the lock here instead instead ; e.g. the operation should not be a no-op because there is a concurrent operation in progress.